// vue-simple-custom-select/dist/vue-simple-select.esm.js
import Vue from 'vue';

function normalizeComponent(template, style, script, scopeId, isFunctionalTemplate, moduleIdentifier /* server only */, shadowMode, createInjector, createInjectorSSR, createInjectorShadow) {
    if (typeof shadowMode !== 'boolean') {
        createInjectorSSR = createInjector;
        createInjector = shadowMode;
        shadowMode = false;
    }
    // Vue.extend constructor export interop.
    const options = typeof script === 'function' ? script.options : script;
    // render functions
    if (template && template.render) {
        options.render = template.render;
        options.staticRenderFns = template.staticRenderFns;
        options._compiled = true;
        // functional template
        if (isFunctionalTemplate) {
            options.functional = true;
        }
    }
    // scopedId
    if (scopeId) {
        options._scopeId = scopeId;
    }
    let hook;
    if (moduleIdentifier) {
        // server build
        hook = function (context) {
            // 2.3 injection
            context =
                context || // cached call
                    (this.$vnode && this.$vnode.ssrContext) || // stateful
                    (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext); // functional
            // 2.2 with runInNewContext: true
            if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
                context = __VUE_SSR_CONTEXT__;
            }
            // inject component styles
            if (style) {
                style.call(this, createInjectorSSR(context));
            }
            // register component module identifier for async chunk inference
            if (context && context._registeredComponents) {
                context._registeredComponents.add(moduleIdentifier);
            }
        };
        // used by ssr in case component is cached and beforeCreate
        // never gets called
        options._ssrRegister = hook;
    }
    else if (style) {
        hook = shadowMode
            ? function (context) {
                style.call(this, createInjectorShadow(context, this.$root.$options.shadowRoot));
            }
            : function (context) {
                style.call(this, createInjector(context));
            };
    }
    if (hook) {
        if (options.functional) {
            // register for functional component in vue file
            const originalRender = options.render;
            options.render = function renderWithStyleInjection(h, context) {
                hook.call(context);
                return originalRender(h, context);
            };
        }
        else {
            // inject component registration as beforeCreate hook
            const existing = options.beforeCreate;
            options.beforeCreate = existing ? [].concat(existing, hook) : [hook];
        }
    }
    return script;
}

/* script */

/* template */
var __vue_render__$1 = function () {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('svg', {
    staticClass: "w-6 h-6",
    attrs: {
      "fill": "none",
      "stroke": "currentColor",
      "viewBox": "0 0 24 24",
      "xmlns": "http://www.w3.org/2000/svg"
    }
  }, [_c('path', {
    attrs: {
      "stroke-linecap": "round",
      "stroke-linejoin": "round",
      "stroke-width": "2",
      "d": "M19 9l-7 7-7-7"
    }
  })]);
};

var __vue_staticRenderFns__$1 = [];
/* style */

const __vue_inject_styles__$1 = undefined;
/* scoped */

const __vue_scope_id__$1 = undefined;
/* module identifier */

const __vue_module_identifier__$1 = undefined;
/* functional template */

const __vue_is_functional_template__$1 = false;
/* style inject */

/* style inject SSR */

/* style inject shadow dom */

const __vue_component__$1 = /*#__PURE__*/normalizeComponent({
  render: __vue_render__$1,
  staticRenderFns: __vue_staticRenderFns__$1
}, __vue_inject_styles__$1, {}, __vue_scope_id__$1, __vue_is_functional_template__$1, __vue_module_identifier__$1, false, undefined, undefined, undefined);

const isOption = option => {
  return typeof option === 'object' && !Array.isArray(option) && 'value' in option && 'label' in option;
};
const isMultipleOptions = options => {
  return Array.isArray(options) && options.every(option => isOption(option));
};
const _uid = () => {
  const generate = () => {
    return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
  };

  return `${generate()}-${generate()}-${generate()}-${generate()}`;
};

var clickOutside = {
  bind(el, binding, vnode) {
    el.clickOutsideEvent = event => {
      if (!(el === event.target || el.contains(event.target))) {
        vnode.context[binding.expression](event);
      }
    };

    document.body.addEventListener('click', el.clickOutsideEvent);
  },

  unbind(el) {
    document.body.removeEventListener('click', el.clickOutsideEvent);
  }

};

var script = /*#__PURE__*/Vue.extend({
  name: 'VueSimpleSelect',
  components: {
    IconChevron: __vue_component__$1
  },

  data() {
    return {
      /**
       * Unique component ID
       */
      uid: _uid(),

      /**
       * If the dropdown is shown or not
       */
      show: false,

      /**
       * The currently selected option, if not in multiple mode
       */
      selectedOption: {},

      /**
       * The currently selected options, if in multiple mode
       */
      selectedOptions: []
    };
  },

  directives: {
    /**
     * Allows to close the dropdown when clicking outside
     */
    'click-outside': { ...clickOutside
    }
  },
  model: {
    /**
     * We bind the v-model to the prop "value" and to the "change" event
     */
    prop: 'value',
    event: 'change'
  },
  props: {
    /**
     * The value to be displayed when nothing is selected
     */
    placeholder: {
      type: String,
      default: ''
    },

    /**
     * The name of the transition to use when showing or hiding the dropdowm.
     */
    transition: {
      type: String,
      default: 'slide-fade'
    },

    /**
     * If the select should not be closed when clicking outside
     */
    noClickOutside: {
      type: Boolean,
      default: false
    },

    /**
     * Ability to place a title on the select button
     */
    title: {
      type: String,
      default: ''
    },

    /**
     * Ability to open the dropdown by default
     */
    opened: {
      type: Boolean,
      default: false
    },

    /**
     * The default value selected
     */
    value: {
      type: [Array, Object],
      default: null,
      required: true
    },

    /**
     * If the select allows multiple items to be selected
     */
    multiple: {
      type: Boolean,
      default: false
    },

    /**
     * Can be used to specify what element labels the button
     */
    labelledby: {
      type: String,
      default: ''
    },

    /**
     * The options displaying inside the dropdown
     */
    options: {
      type: Array,
      required: true,
      default: () => [],
      validator: value => value.every(option => isOption(option))
    }
  },

  mounted() {
    if (this.value === undefined) {
      console.warn('The value should not be undefined');
    } else if (this.multiple && this.value && isOption(this.value)) {
      console.warn('You specified that the select should be multiple, but only gave a single Option as value.');
    }
    /**
     * If we should initialise a single value
     */
    else if (this.value && isOption(this.value)) {
        const value = this.value;
        this.selectedOption = {
          uid: _uid(),
          ...value
        };
      }
      /**
       * If we should initialise multiple values
       */
      else if (this.multiple && this.value && isMultipleOptions(this.value)) {
          const value = this.value;
          const options = value.map(option => ({
            uid: _uid(),
            ...option
          }));
          this.selectedOptions = options;
        }
    /**
     * Opening the dropdown or not
     */


    this.show = this.opened;
  },

  computed: {
    /**
     * Maps unique IDS to each option
     */
    _options() {
      const options = this.options;
      return options.map(option => ({
        uid: _uid(),
        ...option
      }));
    },

    /**
     * Returns the active selected option, for accessibility purposes
     */
    activeDescendant() {
      const option = this.selectedOption;

      if (!this.multiple && option && isOption(option)) {
        return `vue-simple-select-option-${option.uid}`;
      }

      return null;
    }

  },
  methods: {
    /**
     * Returns an Array of every DOM elements corresponding to an select option
     */
    getDOMOptions() {
      const options = Object.entries(this.$refs).map(([key, value]) => {
        return key.includes('vue-simple-select-option') ? value : null;
      }).filter(value => value).flat();
      return options;
    },

    /**
     * Handles navigating with Home and End keys
     */
    gotToOption(event) {
      const options = this.getDOMOptions();

      if (options && options.length) {
        const lastIndex = options.length - 1;

        if (event.code === 'End') {
          options && options.length && options[lastIndex] && options[lastIndex].focus();
        } else if (event.code === 'Home') {
          options && options.length && options[0].focus();
        }
      }
    },

    /**
     * Handles navigating Arrow Down
     */
    gotToNextOption(_) {
      /**
       * Every Options
       */
      const options = this.getDOMOptions();

      if (options && options.length) {
        /**
         * Currently focused element
         */
        const currentElement = document.activeElement;
        const isCurrentElementOption = currentElement && currentElement.id.includes('vue-simple-select-option');
        /**
         * If it is an option
         */

        if (isCurrentElementOption) {
          /**
           * Find the next one and focus it
           */
          const elementIndex = options.findIndex(element => element.id === (currentElement && currentElement.id));

          if (elementIndex > -1) {
            const nextElementIndex = elementIndex + 1;
            const nextElement = options[nextElementIndex];

            if (nextElement) {
              nextElement.focus();
            }
          }
        } else {
          /**
           * Focus the first one
           */
          options && options[0] && options[0].focus();
        }
      }
    },

    /**
     * Handles navigating Arrow Up
     */
    gotToPreviousOption(_) {
      /**
       * Currently focused element
       */
      const options = this.getDOMOptions();

      if (options && options.length) {
        /**
         * Currently focused element
         */
        const currentElement = document.activeElement;
        const isCurrentElementOption = currentElement && currentElement.id.includes('vue-simple-select-option');
        /**
         * If it is an option
         */

        if (isCurrentElementOption) {
          /**
           * Find the previous one and focus it
           */
          const elementIndex = options.findIndex(element => element.id === (currentElement && currentElement.id));

          if (elementIndex > -1) {
            const previousElementIndex = elementIndex - 1;
            const previousElement = options[previousElementIndex];

            if (previousElement) {
              previousElement.focus();
            }
          }
        }
      }
    },

    /**
     * Handles the clicking outside
     */
    handleOutside() {
      if (!this.noClickOutside) {
        this.hideSelect();
      }
    },

    /**
     * Shows the dropdown
     */
    showSelect() {
      this.show = true;
      this.$emit('show');
    },

    /**
     * Hides the dropdown
     */
    hideSelect() {
      this.show = false;
      this.$emit('hide');
    },

    /**
     * Handles seleting a value
     */
    selectValue(option) {
      if (this.multiple) {
        if (this.isSelected(option)) {
          this.selectedOptions = this.selectedOptions.filter(_option => _option.uid !== option.uid);
        } else {
          this.selectedOptions.push(option);
        }

        const options = this.selectedOptions.map(option => this.parseOption(option));
        this.$emit('change', options);
      } else {
        this.selectedOption = option;
        this.$emit('change', this.parseOption(option));
        this.hideSelect();
      }
    },

    /**
     * Handles toggling between showing and hiding the dropdown
     */
    handleClick() {
      this.show ? this.hideSelect() : this.showSelect();
    },

    /**
     * Deletes the uid from the option if it exists
     */
    parseOption(option) {
      if (option) {
        if (option.uid) {
          const duplicate = { ...option
          };
          delete duplicate.uid;
          return duplicate;
        }

        return option;
      }
    },

    /**
     * If the given option is Selected
     */
    isSelected(option) {
      if (option && !this.multiple && this.selectedOption) {
        return this.selectedOption.uid === option.uid;
      } else if (option && this.multiple && this.selectedOptions.length) {
        return !!this.selectedOptions.find(_option => _option.uid === option.uid);
      }

      return false;
    }

  }
});

const isOldIE = typeof navigator !== 'undefined' &&
    /msie [6-9]\\b/.test(navigator.userAgent.toLowerCase());
function createInjector(context) {
    return (id, style) => addStyle(id, style);
}
let HEAD;
const styles = {};
function addStyle(id, css) {
    const group = isOldIE ? css.media || 'default' : id;
    const style = styles[group] || (styles[group] = { ids: new Set(), styles: [] });
    if (!style.ids.has(id)) {
        style.ids.add(id);
        let code = css.source;
        if (css.map) {
            // https://developer.chrome.com/devtools/docs/javascript-debugging
            // this makes source maps inside style tags work properly in Chrome
            code += '\n/*# sourceURL=' + css.map.sources[0] + ' */';
            // http://stackoverflow.com/a/26603875
            code +=
                '\n/*# sourceMappingURL=data:application/json;base64,' +
                    btoa(unescape(encodeURIComponent(JSON.stringify(css.map)))) +
                    ' */';
        }
        if (!style.element) {
            style.element = document.createElement('style');
            style.element.type = 'text/css';
            if (css.media)
                style.element.setAttribute('media', css.media);
            if (HEAD === undefined) {
                HEAD = document.head || document.getElementsByTagName('head')[0];
            }
            HEAD.appendChild(style.element);
        }
        if ('styleSheet' in style.element) {
            style.styles.push(code);
            style.element.styleSheet.cssText = style.styles
                .filter(Boolean)
                .join('\n');
        }
        else {
            const index = style.ids.size - 1;
            const textNode = document.createTextNode(code);
            const nodes = style.element.childNodes;
            if (nodes[index])
                style.element.removeChild(nodes[index]);
            if (nodes.length)
                style.element.insertBefore(textNode, nodes[index]);
            else
                style.element.appendChild(textNode);
        }
    }
}

/* script */
const __vue_script__ = script;
/* template */

var __vue_render__ = function () {
  var _vm = this;

  var _h = _vm.$createElement;

  var _c = _vm._self._c || _h;

  return _c('div', {
    directives: [{
      name: "click-outside",
      rawName: "v-click-outside",
      value: _vm.handleOutside,
      expression: "handleOutside"
    }],
    staticClass: "vue-simple-select-container",
    on: {
      "keyup": [function ($event) {
        if (!$event.type.indexOf('key') && _vm._k($event.keyCode, "down", 40, $event.key, ["Down", "ArrowDown"])) {
          return null;
        }

        return _vm.gotToNextOption($event);
      }, function ($event) {
        if (!$event.type.indexOf('key') && _vm._k($event.keyCode, "up", 38, $event.key, ["Up", "ArrowUp"])) {
          return null;
        }

        return _vm.gotToPreviousOption($event);
      }, _vm.gotToOption]
    }
  }, [_vm._t("button", [_c('button', {
    staticClass: "vue-simple-select-button",
    attrs: {
      "title": _vm.title,
      "aria-haspopup": "listbox",
      "aria-expanded": _vm.show,
      "aria-owns": "dropdown-" + _vm.uid,
      "aria-labelledby": _vm.labelledby,
      "aria-activedescendant": _vm.activeDescendant
    },
    on: {
      "click": _vm.handleClick
    }
  }, [_vm._t("inner-button"), _vm._v(" "), _vm.placeholder && (!_vm.selectedOption || !Object.keys(_vm.selectedOption).length) && (!_vm.selectedOptions || !_vm.selectedOptions.length) ? _c('span', {
    staticClass: "vue-simple-select-placeholder"
  }, [_vm._v("\n        " + _vm._s(_vm.placeholder) + "\n      ")]) : _vm._e(), _vm._v(" "), _vm.selectedOptions && _vm.multiple && _vm.selectedOptions.length ? _c('span', {
    staticClass: "vue-simple-select-selected"
  }, [_c('span', {
    staticClass: "vue-simple-select-label"
  }, [_vm._v(_vm._s(_vm.selectedOptions.length) + " ")]), _vm._v("selected\n      ")]) : _vm.selectedOption ? _c('span', {
    staticClass: "vue-simple-select-selected-option"
  }, [_vm._v("\n        " + _vm._s(_vm.selectedOption.label) + "\n      ")]) : _vm._e(), _vm._v(" "), _vm._t("before-icon"), _vm._v(" "), _vm._t("icon", [_c('icon-chevron', {
    staticClass: "vue-simple-select-icon",
    class: {
      reverse: _vm.show
    }
  })], null, {
    show: _vm.show
  }), _vm._v(" "), _vm._t("after-icon")], 2)], null, {
    handleClick: _vm.handleClick,
    showSelect: _vm.showSelect,
    hideSelect: _vm.hideSelect
  }), _vm._v(" "), _c('transition', {
    attrs: {
      "name": _vm.transition
    }
  }, [_vm.show ? _vm._t("dropdown", [_c('div', {
    staticClass: "vue-simple-select-dropdown",
    attrs: {
      "id": "dropdown-" + _vm.uid
    }
  }, [_vm._t("dropdown-content", [_c('ul', {
    ref: "options-list",
    staticClass: "vue-simple-select-list",
    attrs: {
      "role": "listbox"
    }
  }, _vm._l(_vm._options, function (option, index) {
    return _c('li', {
      key: "option-" + index,
      staticClass: "vue-simple-select-list-item"
    }, [_vm._t("single-option", [_c('button', {
      ref: "vue-simple-select-option-" + option.uid,
      refInFor: true,
      staticClass: "vue-simple-select-option",
      class: {
        selected: _vm.isSelected(option)
      },
      attrs: {
        "id": "vue-simple-select-option-" + option.uid,
        "role": "option"
      },
      on: {
        "click": function ($event) {
          return _vm.selectValue(option);
        }
      }
    }, [_vm._v("\n                  " + _vm._s(option.label) + "\n                ")])], null, {
      option: option,
      index: index,
      selectValue: _vm.selectValue,
      isSelected: _vm.isSelected
    })], 2);
  }), 0)], null, {
    selectValue: _vm.selectValue,
    isSelected: _vm.isSelected
  })], 2)], null, {
    selectValue: _vm.selectValue,
    isSelected: _vm.isSelected,
    show: _vm.show
  }) : _vm._e()], 2)], 2);
};

var __vue_staticRenderFns__ = [];
/* style */

const __vue_inject_styles__ = function (inject) {
  if (!inject) return;
  inject("data-v-46be17c1_0", {
    source: "*{--tw-shadow:0 0 transparent;--tw-ring-inset:var(--tw-empty, );/*!*//*!*/--tw-ring-offset-width:0px;--tw-ring-offset-color:#fff;--tw-ring-color:rgba(59,130,246,0.5);--tw-ring-offset-shadow:0 0 transparent;--tw-ring-shadow:0 0 transparent}@keyframes vss-spin{to{transform:rotate(1turn)}}@keyframes vss-ping{75%,to{transform:scale(2);opacity:0}}@keyframes vss-pulse{50%{opacity:.5}}@keyframes vss-bounce{0%,to{transform:translateY(-25%);animation-timing-function:cubic-bezier(.8,0,1,1)}50%{transform:none;animation-timing-function:cubic-bezier(0,0,.2,1)}}.vue-simple-select-container{font-family:system-ui,-apple-system,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;/*! tailwindcss v2.1.2 | MIT License | https://tailwindcss.com *//*! modern-normalize v1.0.0 | MIT License | https://github.com/sindresorhus/modern-normalize */min-width:8rem;position:relative;--tw-text-opacity:1;color:rgba(31,41,55,var(--tw-text-opacity))}.vue-simple-select-container .slide-fade-enter-active,.vue-simple-select-container .slide-fade-leave-active{transition:all .3s ease}.vue-simple-select-container .slide-fade-enter,.vue-simple-select-container .slide-fade-leave-to{transform:translateY(-10px);opacity:0}.vue-simple-select-container hr{height:0;color:inherit}.vue-simple-select-container abbr[title]{-webkit-text-decoration:underline dotted;text-decoration:underline dotted}.vue-simple-select-container b,.vue-simple-select-container strong{font-weight:bolder}.vue-simple-select-container code,.vue-simple-select-container kbd,.vue-simple-select-container pre,.vue-simple-select-container samp{font-family:ui-monospace,SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace;font-size:1em}.vue-simple-select-container small{font-size:80%}.vue-simple-select-container sub,.vue-simple-select-container sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}.vue-simple-select-container sub{bottom:-.25em}.vue-simple-select-container sup{top:-.5em}.vue-simple-select-container table{text-indent:0;border-color:inherit}.vue-simple-select-container button,.vue-simple-select-container input,.vue-simple-select-container optgroup,.vue-simple-select-container select,.vue-simple-select-container textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0}.vue-simple-select-container button,.vue-simple-select-container select{text-transform:none}.vue-simple-select-container [type=button],.vue-simple-select-container [type=reset],.vue-simple-select-container [type=submit],.vue-simple-select-container button{-webkit-appearance:button}.vue-simple-select-container ::-moz-focus-inner{border-style:none;padding:0}.vue-simple-select-container :-moz-focusring{outline:1px dotted ButtonText}.vue-simple-select-container :-moz-ui-invalid{box-shadow:none}.vue-simple-select-container legend{padding:0}.vue-simple-select-container progress{vertical-align:baseline}.vue-simple-select-container ::-webkit-inner-spin-button,.vue-simple-select-container ::-webkit-outer-spin-button{height:auto}.vue-simple-select-container [type=search]{-webkit-appearance:textfield;outline-offset:-2px}.vue-simple-select-container ::-webkit-search-decoration{-webkit-appearance:none}.vue-simple-select-container ::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}.vue-simple-select-container summary{display:list-item}.vue-simple-select-container blockquote,.vue-simple-select-container dd,.vue-simple-select-container dl,.vue-simple-select-container figure,.vue-simple-select-container h1,.vue-simple-select-container h2,.vue-simple-select-container h3,.vue-simple-select-container h4,.vue-simple-select-container h5,.vue-simple-select-container h6,.vue-simple-select-container hr,.vue-simple-select-container p,.vue-simple-select-container pre{margin:0}.vue-simple-select-container button{background-color:transparent;background-image:none}.vue-simple-select-container button:focus{outline:1px dotted;outline:5px auto -webkit-focus-ring-color}.vue-simple-select-container fieldset{margin:0;padding:0}.vue-simple-select-container ol,.vue-simple-select-container ul{list-style:none;margin:0;padding:0}.vue-simple-select-container html{font-family:ui-sans-serif,system-ui,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Arial,Noto Sans,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;line-height:1.5}.vue-simple-select-container body{font-family:inherit;line-height:inherit}.vue-simple-select-container *,.vue-simple-select-container :after,.vue-simple-select-container :before{box-sizing:border-box;border:0 solid #e5e7eb}.vue-simple-select-container hr{border-top-width:1px}.vue-simple-select-container img{border-style:solid}.vue-simple-select-container textarea{resize:vertical}.vue-simple-select-container input:-ms-input-placeholder,.vue-simple-select-container textarea:-ms-input-placeholder{opacity:1;color:#9ca3af}.vue-simple-select-container input::placeholder,.vue-simple-select-container textarea::placeholder{opacity:1;color:#9ca3af}.vue-simple-select-container [role=button],.vue-simple-select-container button{cursor:pointer}.vue-simple-select-container table{border-collapse:collapse}.vue-simple-select-container h1,.vue-simple-select-container h2,.vue-simple-select-container h3,.vue-simple-select-container h4,.vue-simple-select-container h5,.vue-simple-select-container h6{font-size:inherit;font-weight:inherit}.vue-simple-select-container a{color:inherit;text-decoration:inherit}.vue-simple-select-container button,.vue-simple-select-container input,.vue-simple-select-container optgroup,.vue-simple-select-container select,.vue-simple-select-container textarea{padding:0;line-height:inherit;color:inherit}.vue-simple-select-container code,.vue-simple-select-container kbd,.vue-simple-select-container pre,.vue-simple-select-container samp{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace}.vue-simple-select-container audio,.vue-simple-select-container canvas,.vue-simple-select-container embed,.vue-simple-select-container iframe,.vue-simple-select-container img,.vue-simple-select-container object,.vue-simple-select-container svg,.vue-simple-select-container video{display:block;vertical-align:middle}.vue-simple-select-container img,.vue-simple-select-container video{max-width:100%;height:auto}.vue-simple-select-container .vue-simple-select-button{--tw-border-opacity:1;border-color:rgba(243,244,246,var(--tw-border-opacity));border-width:1px;display:flex;align-items:center;padding:.5rem;--tw-shadow:0 1px 2px 0 rgba(0,0,0,0.05)}.vue-simple-select-container .vue-simple-select-button,.vue-simple-select-container .vue-simple-select-dropdown{border-radius:.25rem;min-width:100%;box-shadow:var(--tw-ring-offset-shadow,0 0 transparent),var(--tw-ring-shadow,0 0 transparent),var(--tw-shadow);width:max-content}.vue-simple-select-container .vue-simple-select-dropdown{padding-top:.25rem;padding-bottom:.25rem;position:absolute;left:0;top:100%;--tw-shadow:0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06)}.vue-simple-select-container .vue-simple-select-option:hover{--tw-bg-opacity:1;background-color:rgba(243,244,246,var(--tw-bg-opacity))}.vue-simple-select-container .vue-simple-select-option{padding:.5rem .75rem;text-align:left;width:100%;transition-property:background-color,border-color,color,fill,stroke;transition-timing-function:cubic-bezier(.4,0,.2,1);transition-duration:.15s}.vue-simple-select-container .vue-simple-select-option.selected{--tw-bg-opacity:1;background-color:rgba(219,234,254,var(--tw-bg-opacity));font-weight:500;--tw-text-opacity:1;color:rgba(37,99,235,var(--tw-text-opacity))}.vue-simple-select-container .vue-simple-select-icon{height:1rem;margin-left:auto;width:1rem;--tw-translate-x:0;--tw-translate-y:0;--tw-rotate:0;--tw-skew-x:0;--tw-skew-y:0;--tw-scale-x:1;--tw-scale-y:1;transform:translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));transition-property:transform;transition-timing-function:cubic-bezier(.4,0,.2,1);transition-duration:.15s}.vue-simple-select-container .vue-simple-select-icon.reverse{--tw-rotate:180deg}.vue-simple-select-container .vue-simple-select-placeholder,.vue-simple-select-container .vue-simple-select-selected{--tw-text-opacity:1;color:rgba(107,114,128,var(--tw-text-opacity))}.vue-simple-select-container .vue-simple-select-label{font-weight:700;--tw-text-opacity:1;color:rgba(75,85,99,var(--tw-text-opacity))}",
    map: undefined,
    media: undefined
  });
};
/* scoped */


const __vue_scope_id__ = undefined;
/* module identifier */

const __vue_module_identifier__ = undefined;
/* functional template */

const __vue_is_functional_template__ = false;
/* style inject SSR */

/* style inject shadow dom */

const __vue_component__ = /*#__PURE__*/normalizeComponent({
  render: __vue_render__,
  staticRenderFns: __vue_staticRenderFns__
}, __vue_inject_styles__, __vue_script__, __vue_scope_id__, __vue_is_functional_template__, __vue_module_identifier__, false, createInjector, undefined, undefined);

// Import vue component

// Default export is installable instance of component.
// IIFE injects install function into component, allowing component
// to be registered via Vue.use() as well as Vue.component(),
var entry_esm = /*#__PURE__*/(() => {
  // Assign InstallableComponent type
  const installable = __vue_component__; // Attach install function executed by Vue.use()

  installable.install = Vue => {
    Vue.component('VueSimpleSelect', installable);
  };

  return installable;
})(); // It's possible to expose named exports when writing components that can
// also be used as directives, etc. - eg. import { RollupDemoDirective } from 'rollup-demo';
// export const RollupDemoDirective = directive;

export default entry_esm;
